packages/server/src/subscriptions/websockets.ts

```diff
@@ -51,7 +61,35 @@ async function setupSubscriptionHandler(): Promise<void> {
     for (const [resource, subscriptionId, options] of subEventArgsArr) {
       const bundle = createSubEventNotification(resource, subscriptionId, options);
       for (const socket of subToWsLookup.get(subscriptionId) ?? []) {
-        socket.send(JSON.stringify(bundle), { binary: false });
+        // Get the repo for this socket in the context of the subscription
+        const subMetadataMap = wsToSubLookup.get(socket);
+        if (!subMetadataMap) {
+          // We should really never hit this log, this is a true error
+          globalLogger.error('[WS] Unable to find sub metadata map for WebSocket');
```

Comment:
Hm.  should we kill the subscription in this case?

Reply:
I don't think we should kill the subscription, it could just be that the socket disconnected and for some reason this didn't get cleaned up. We don't support multiple WebSockets binding to the same subscription yet but in the future we could. I would like to see if we even hit this case in the logs first 👀